server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -891,6 +882,10 @@ public int getNumberOfReplicas() {
         return settings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, null);
     }
 
+    public boolean isSegrepEnabled() {
```

Comment:
Nitpick - I'd prefer the method name to be explicit i.e. `isSegmentReplicationEnabled`